src/models/Expense.model.js

```diff
@@ -1,11 +1,123 @@
 'use strict';
 
 const { sequelize } = require('../db.js');
+const usersService = require('../services/user');
+const { Op, DataTypes } = require('sequelize');
 
 const Expense = sequelize.define(
-  // your code goes here
+  'Expense',
+  {
+    id: {
+      type: DataTypes.UUID,
+      defaultValue: DataTypes.UUIDV4,
+      primaryKey: true,
+    },
+    userId: {
+      type: DataTypes.UUID,
+      allowNull: false,
+    },
+    category: {
+      type: DataTypes.STRING,
+    },
+    title: {
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+    amount: {
+      type: DataTypes.INTEGER,
+      allowNull: false,
+    },
+    spentAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+    },
+    note: {
+      type: DataTypes.STRING,
+    },
+  },
+  {
+    tableName: 'expense',
+  },
 );
 
+const init = async () => {
+  Expense.sync({ force: true });
+};
+
+const getAll = async (userId, categories, from, to) => {
+  const filter = {};
+
+  if (userId) {
+    filter.userId = userId;
+  }
+
+  if (categories) {
+    filter.category = categories;
+  }
+
+  if (from && !to) {
+    filter.spentAt = { [Op.gte]: from };
+  }
+
+  return Expense.findAll({
+    where: filter,
+  });
+};
+
+const getById = async (id) => {
+  return Expense.findByPk(id);
+};
+
+const create = async (data) => {
+  try {
+    const result = await sequelize.transaction(async (t) => {
+      const user = usersService.getById(data.userId);
+
+      if (!user) {
+        return null;
+      }
+
+      return Expense.create(data);
+    });
+
+    return result;
+  } catch (error) {
+    throw error;
```

Comment:
Consider making a response with a 500 error code if there are some issues with the database